File: search_all_strings.py
import re
import json

file_path = r"C:\Users\jahan\.gemini\antigravity\brain\42c65725-4574-4d8b-8022-d0909dd83905\.system_generated\steps\56\content.md"

with open(file_path, 'r', encoding='utf-8') as f:
    content = f.read()

# Let's search for "timeline" or "media" in a case-insensitive way
print("Scanning for timeline media keys...")
matches = re.findall(r'"(edge_owner_to_timeline_media|edge_saved_media|timeline_media)"', content)
print("Matches found:", matches)

# Let's search for strings that look like captions or descriptions
# Instagram JSON structure: {"node": {"text": "..."}}
# Let's look for "node" structures containing "text"
# Let's search for the word "node" and print the next 200 characters
nodes = re.findall(r'"node"\s*:\s*\{([^}]+)\}', content)
print("Found 'node' blocks:", len(nodes))
for i, node in enumerate(nodes[:20]):
    print(f"Node {i}: {node[:120]}")

# Let's search for any RERA number, phone, email, price or location in the whole file
print("\nScanning for project keywords in the entire file:")
keywords = ["rera", "pr/gj", "Vaishno", "Circle", "Gota", "Ahmedabad", "BHK", "Developer", "Lakh", "Crore", "price", "contact", "phone", "email"]
for kw in keywords:
    matches_kw = re.findall(rf'([^"\n]*?{kw}[^"\n]*)', content, re.IGNORECASE)
    if matches_kw:
        print(f"Keyword '{kw}' matches ({len(matches_kw)}):")
        for m in list(set(matches_kw))[:5]:
            print("  -", m.strip()[:150])

File: search_all_strings.js
const fs = require('fs');
const path = require('path');

const filePath = 'C:\\Users\\jahan\\.gemini\\antigravity\\brain\\42c65725-4574-4d8b-8022-d0909dd83905\\.system_generated\\steps\\56\\content.md';

try {
  if (!fs.existsSync(filePath)) {
    console.error('File not found:', filePath);
    process.exit(1);
  }

  const content = fs.readFileSync(filePath, 'utf8');
  console.log('File size:', content.length, 'bytes');

  const keywords = ["rera", "pr/gj", "Vaishno", "Circle", "Gota", "Ahmedabad", "BHK", "Developer", "Lakh", "Crore", "price", "contact", "phone", "email"];
  
  console.log('\n--- Scanning for keywords (Safe Indexing) ---');
  keywords.forEach(kw => {
    let idx = -1;
    let count = 0;
    console.log(`\nMatches for "${kw}":`);
    while (true) {
      idx = content.toLowerCase().indexOf(kw.toLowerCase(), idx + 1);
      if (idx === -1) break;
      count++;
      if (count <= 3) {
        // extract context of 120 chars around it
        const start = Math.max(0, idx - 50);
        const end = Math.min(content.length, idx + 100);
        const chunk = content.substring(start, end).replace(/\n/g, ' ').trim();
        console.log(`  - Index ${idx}: ...${chunk}...`);
      }
    }
    console.log(`Total count: ${count}`);
  });

} catch (err) {
  console.error('Error reading file:', err);
}
